router/core/operation_processor.go

```diff
@@ -372,10 +372,25 @@ func (o *OperationKit) FetchPersistedOperation(ctx context.Context, clientInfo *
 	persistedOperationData, err := o.operationProcessor.persistedOperationClient.PersistedOperation(ctx, clientInfo.Name, o.parsedOperation.GraphQLRequestExtensions.PersistedQuery.Sha256Hash)
 	if err != nil {
 		return false, err
+	} else if persistedOperationData == nil && o.parsedOperation.Request.Query == "" {
+		// If the client has APQ enabled, throw an error if the operation wasn't attached to the request
+		return false, &persistedoperation.PersistentOperationNotFoundError{
+			ClientName: clientInfo.Name,
+			Sha256Hash: o.parsedOperation.GraphQLRequestExtensions.PersistedQuery.Sha256Hash,
+		}
 	}
 	// it's important to make a copy of the persisted operation data, because it's used in the cache
 	// we might modify it later, so we don't want to modify the cached data
-	o.parsedOperation.Request.Query = string(persistedOperationData)
+	if persistedOperationData != nil {
+		o.parsedOperation.Request.Query = string(persistedOperationData)
+	}
+
+	// If the operation was fetched with APQ, save it again to renew the TTL
+	if o.operationProcessor.persistedOperationClient.ApqEnabled() {
+		if err = o.operationProcessor.persistedOperationClient.SaveOperation(ctx, clientInfo.Name, o.parsedOperation.GraphQLRequestExtensions.PersistedQuery.Sha256Hash, o.parsedOperation.Request.Query); err != nil {
```

Comment:
Will this mean that we're saving every request to the cache when APQ is enabled? This might result in excessive writes to the cache.

Reply:
It would. I can change the `Client` interface so that it will also return the source, which could be useful